### PR TITLE
v107.0.0

### PR DESCRIPTION
**Pull request by codeman7**
# 107.0.0

In this major release we deleted the deprecated themer classes: `MDCAlertScheme` and `MDCAlertControllerThemer`.

## Breaking changes

Deleted `MDCAlertScheme` and `MDCAlertControllerThemer` class, please use the theming extensions.
Learn more at https://github.com/material-components/material-components-ios/blob/develop/docs/theming.md#migration-guide-themers-to-theming-extensions

## New features

In this release we added the ability to be informed about presentation events of a `MDCAlertController` as well as various minor improvements to the `Dialogs` component and other bug fixes.

### Dialogs

*new* property: `delegate` in `MDCAlertController`

*new* protocol: `MDCAlertControllerDelegate`

### BottomSheet

* [Add example to reproduce issue 9773 (#9789)](https://github.com/material-components/material-components-ios/commit/00a4f37a4aae8dd5e9e5e66f7a8315aefac57593) (Bryan Oltman)
* [KVO `contentInset` property (#9726)](https://github.com/material-components/material-components-ios/commit/b027eeed70db247274aaaaa1ecfe9a85db5243c0) (Robert Moore)

### Chips

* [Add snapshot tests for MDCChipCollectionViewFlowLayout (#9796)](https://github.com/material-components/material-components-ios/commit/45952d95a7bfd9a9da99318e69f05c4065880375) (Bryan Oltman)
* [Refactor MDCChipCollectionViewFlowLayout (#9799)](https://github.com/material-components/material-components-ios/commit/fbc35415a2f9342102a003cdefea28a4f955dfda) (Bryan Oltman)

### Dialogs

* [Adding actions alignment API in a private header. (#9795)](https://github.com/material-components/material-components-ios/commit/8a8077d20eaf511a3a8f8e78b63ec7ced4b857a8) (Galia Kaufman)
* [Adding alignment snapshot tests (#9801)](https://github.com/material-components/material-components-ios/commit/64a79b29bf7ceba090f65520caf49f5bd56ec38a) (Galia Kaufman)
* [Adding justified actions alignment calculations to actions size calculations (#9800)](https://github.com/material-components/material-components-ios/commit/3537e43ef82c549a7c449a7fb6bdfbea829092ce) (Galia Kaufman)
* [Delete deprecated DialogThemer (#9776)](https://github.com/material-components/material-components-ios/commit/511da587d485fac0407ccc8a0519aaa941dceea5) (Bryan Oltman)
* [Moving actions layout out to simplify layoutSubviews. (#9803)](https://github.com/material-components/material-components-ios/commit/fa919c8da817fa59ba6bdd6379454dd061846886) (Galia Kaufman)
* [Support impressions logging in Material Design Dialogs. (#9794)](https://github.com/material-components/material-components-ios/commit/c2af4ae17f0feec12a1a6778dba3bb80505263c7) (Galia Kaufman)
* [Use consistent syntax for float constants (#9750)](https://github.com/material-components/material-components-ios/commit/e96b7efd28d7284eb090dab1408e0083dc06353e) (Galia Kaufman)

## TextField

* [implement resignFirstResponder in MDCMultilineTextField (#9793)](https://github.com/material-components/material-components-ios/commit/d83af534c3121bc8f9f2c0cb459917bec33158fe) (Bryan Oltman)

### CODE REVIEW

**Review by iberezan — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,52 @@
+# 107.0.0
+
+In this major release we deleted the deprecated themer classes: `MDCAlertScheme` and `MDCAlertControllerThemer`.
+
+## Breaking changes
+
+Deleted `MDCAlertScheme` and `MDCAlertControllerThemer` class, please use the theming extensions.
```
Comment (galiak11) [inline CHANGELOG.md:7]: Should we be more specific about what to use instead? How about something like:
"please use `MDCAlertController (MaterialTheming)` extension instead."

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+
+In this release we added the ability to be informed about presentation events of a `MDCAlertController` as well as various minor improvements to the `Dialogs` component and other bug fixes.
+
+## API changes
+
+### Dialogs
+
+*new* property: `delegate` in `MDCAlertController`
+
+*new* protocol: `MDCAlertControllerDelegate`
```
Comment (galiak11) [inline CHANGELOG.md]: You can add something along these lines:
"The new delegate MDCAlertControllerDelegate protocol exposes dialogs life cycle events that are useful for impression logging."

**Review by galiak11 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (codeman7) [reply CHANGELOG.md:7]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (codeman7) [reply CHANGELOG.md]: We typically don't describe the APIs functionality in the API section, I've updated the _new features_ to be along these lines.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,52 @@
+# 107.0.0
+
+In this major release we deleted the deprecated themer classes: `MDCAlertScheme` and `MDCAlertControllerThemer`.
```
Comment (andrewoverton) [inline CHANGELOG.md:3]: How about this:
> In this major release we deleted the deprecated `MDCAlertScheme` and `MDCAlertControllerThemer ` classes.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,52 @@
...
+
+## Breaking changes
+
+Deleted `MDCAlertScheme` and `MDCAlertControllerThemer` class, please use `MDCAlertController (MaterialTheming)` extension instead.
```
Comment (andrewoverton) [inline CHANGELOG.md]: `MDCAlertScheme` and `MDCAlertControllerThemer` have been deleted. Please use the `MDCAlertController (MaterialTheming)` extension instead.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+
+## New features
+
+The new delegate MDCAlertControllerDelegate protocol exposes dialogs life cycle events that are useful for impression logging."
```
Comment (andrewoverton) [inline CHANGELOG.md:11]: Should Dialogs be capitalized? Also, it looks like there's a closing quotation mark at the end of the sentence with no opening quotation mark before it.

Comment (andrewoverton) [reply CHANGELOG.md:11]: >The new MDCAlertControllerDelegate protocol exposes Dialogs life cycle events that are useful for impression logging.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,52 @@
...
+In this release we added a new `delegate` property to `MDCAlertController` which exposes dialogs lifecycle events that are useful for impression logging. As well as various minor improvements to the `Dialogs` component and other bug fixes.
```
Comment (andrewoverton) [inline CHANGELOG.md:12]: This sentence seems partially redundant. What about this:
>`MDCAlertController` now has a `delegate` property conforming to the new `MDCAlertControllerDelegate` property.

**Review by andrewoverton — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (codeman7) [reply CHANGELOG.md:11]: This has been deleted.

File: CHANGELOG.md (hunk 6 above)
Comment (codeman7) [reply CHANGELOG.md:12]: Forgot to delete the first sentence sorry.

**Pull request merged**